compiler/noirc_frontend/src/tests.rs

```diff
@@ -1944,17 +1987,24 @@ fn implicit_numeric_generics_elaborator() {
     }
     "#;
     let errors = get_program_errors(src);
-    assert_eq!(errors.len(), 4);
-
-    for error in errors.iter() {
-        if let CompilationError::ResolverError(ResolverError::UseExplicitNumericGeneric { ident }) =
-            &errors[0].0
-        {
-            assert!(matches!(ident.0.contents.as_str(), "MaxLen" | "Len"));
-        } else {
-            panic!("Expected ResolverError::UseExplicitNumericGeneric but got {:?}", error);
-        }
-    }
+    assert_eq!(errors.len(), 3);
+    assert!(matches!(
+        errors[0].0,
+        CompilationError::TypeError(TypeCheckError::TypeKindMismatch { .. }),
+    ));
+    assert!(matches!(
+        errors[1].0,
+        CompilationError::ResolverError(ResolverError::VariableNotDeclared { .. }),
+    ));
+    // N
```

Comment:
Where does `N` come from? This is due to `self.len < MaxLen` right?
```suggestion
```

Reply:
Yes